cmd/zfs/zfs_main.c

```diff
@@ -7237,15 +7256,27 @@ unshare_unmount(int op, int argc, char **argv)
 
 			switch (op) {
 			case OP_SHARE:
+#if defined(__FreeBSD__)
```

Comment:
To minimize changes you can use `zfs_unshareall_bytype()` here, passing `NULL` for `proto` is functionally equivalent.